cpp/example_code/s3/CMakeLists.txt

```diff
@@ -1,49 +1,104 @@
-# Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
-# This file is licensed under the Apache License, Version 2.0 (the "License").
-# You may not use this file except in compliance with the License. A copy of
-# the License is located at
-# http://aws.amazon.com/apache2.0/
-# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
-# CONDITIONS OF ANY KIND, either express or implied. See the License for the
-# specific language governing permissions and limitations under the License.
-
-cmake_minimum_required(VERSION 3.2)
-project(s3-examples)
-set (CMAKE_CXX_STANDARD 11)
-
-# Locate the aws sdk for c++ package.
+﻿# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Set the minimum required version of CMake for this project.
+cmake_minimum_required(VERSION 3.8)
+
+# Set this project's name.
+project("s3-examples")
+
+# Set the C++ standard to use to build this target.
+set(CMAKE_CXX_STANDARD 11)
+
+# Enable CTest for testing these code examples.
+include(CTest)
+
+# Build shared libraries by default.
+if(NOT BUILD_SHARED_LIBS)
+	set(BUILD_SHARED_LIBS ON)
+endif()
+
+# If the compiler is some version of Microsoft Visual C++, or another compiler simulating C++, 
+# and building as shared libraries, then dynamically link to those shared libraries.
+if(MSVC AND BUILD_SHARED_LIBS)
+	add_definitions(-DUSE_IMPORT_EXPORT)
+endif()
+
+# Find the AWS SDK for C++ package.
 find_package(AWSSDK REQUIRED COMPONENTS s3)
 

```

Comment:
I'd change these lines to:
```cmake
# Find the AWS SDK for C++ package.
find_package(AWSSDK REQUIRED COMPONENTS s3)

# If the compiler is some version of Microsoft Visual C++, or another compiler simulating C++,
# and building as shared libraries, then dynamically link to those shared libraries.
if(MSVC AND BUILD_SHARED_LIBS)
	add_definitions(-DUSE_IMPORT_EXPORT)
	# Copy relevant AWS SDK for C++ libraries into the current binary directory for running and debugging.
	list(APPEND SERVICE_LIST s3)
	AWSSDK_CPY_DYN_LIBS(SERVICE_LIST "" ${CMAKE_CURRENT_BINARY_DIR}/Debug)
endif()
```
1. We need to copy DLLs only when linking to dynamic libraries on Windows.
2. we can call `AWSSDK_CPY_DYN_LIBS` only after we succeeds to `find_package(AWSSDK REQUIRED COMPONENTS s3)`
3. The first parameter of `AWSSDK_CPY_DYN_LIBS` should be a list, so I wrap "s3" into a list.
4. Also, changed `${CMAKE_CURRENT_BINARY_DIR}` to `${CMAKE_CURRENT_BINARY_DIR}/Debug` to copy the DLLs to the same directories of executables.